### ISSUE TITLE
Minetest plays stereo sound as mono

### ISSUE THREAD

**Issue by Penguinum** [minetest/minetest#1823]
When I use minetest.sound_play("my_stereo_sound"), Minetest plays this sound as mono. Wiki says: "For positional playing of sounds, only single-channel (mono) files are supported. Otherwise OpenAL will play them non-positionally."
Does this mean that OpenAL will play stereo sounds not only without any "binding" to position, but also as mono? Or is it a bug? Or maybe I'm just doing something wrong?
Update: I noticed that output audio is not mono, but channels are very similar.
(Sorry for my bad English.)

**Comment by sofar** [comment]
OpenAL is designed as a 3d sound mixer, such that sounds coming from the same location can never be stereo, after all the definition of "stereo" is hearing 2 sounds coming from different positions. 2 sounds coming from one and the same location will therefore be "mixed" together.

If you want to convey stereo sound to the player, you will have the following 2 options:

1. use to_player = <playername>, turning it into a location-less sound (which can be stereo). The sound will only play for that player.

2. create 2 separate mono sound streams, and play them in 2 different positions in-game. The player will hear stereo.

Playing 2 sounds in the same location, or attempting to play a stereo stream as a positional stream may have side effects, as the mixing of audio may vary per architecture, and kcat admits it's an approximation from what I've seen (http://openal.org/pipermail/openal/2015-February/000330.html)

TL:DR do not play stereo sounds as positional sounds, it's not supported.

**Comment by sofar** [comment]
@sfan5 considering above comment, I suggest closing this - I do not believe this is a bug.

**Comment by paramat** [comment]
^ @Penguinum
